rust/bridge/shared/src/node/convert.rs

```diff
@@ -632,47 +631,170 @@ pub(crate) unsafe fn extend_lifetime<'a, 'b: 'a, T: ?Sized>(some_ref: &'a T) ->
+    type JsBoxContents = RefCell<T>;
+}
+
+/// Shorthand for `T::Strategy::JsBoxContents` when `T: `[`BridgeHandle`], which Rust can't always
+/// resolve.
+type BoxContentsFor<T> = <<T as BridgeHandle>::Strategy as BridgeHandleStrategy<T>>::JsBoxContents;
+
+impl<'a, T: BridgeHandle> ResultTypeInfo<'a> for T {
+    type ResultType = JsValue;
+    fn convert_into(self, cx: &mut impl Context<'a>) -> NeonResult<Handle<'a, Self::ResultType>> {
+        Ok(cx
+            .boxed(DefaultFinalize(BoxContentsFor::<T>::from(self)))
+            .upcast())
+    }
+}
+/// Accesses a boxed Rust value using a custom reference (such as [`std::cell::RefMut`]).
+///
+/// Once created, the value can be accessed using the usual [`Deref`] and [`DerefMut`] operations.
+///
+/// Also stores the JavaScript handle that owns the JsBox to ensure that the Rust value is kept
+/// alive.
+pub struct BoxedBorrow<'a, Borrow: Deref + 'a>
+where
+    Borrow::Target: BridgeHandle,
+{
+    _owner: Handle<'a, DefaultJsBox<BoxContentsFor<Borrow::Target>>>,
+    borrow: Borrow,
+}
+
+impl<'a, Borrow: Deref + 'a> BoxedBorrow<'a, Borrow>
+where
+    Borrow::Target: BridgeHandle,
+{
+    /// Creates a BoxedBorrow by accessing `wrapper`, assuming it does in fact reference a boxed
+    /// Rust value under the `_nativeHandle` property.
+    ///
+    /// The `borrow` callback will be used to get an active reference for the Rust value.
+    /// For example, if the "box contents" for this handle use a [`RefCell`], `borrow` might
+    /// be [`RefCell::borrow_mut`] to get a mutable reference out. For a plain value, `borrow`
+    /// can be the identity function.
+    pub fn from_wrapper(
+        cx: &mut impl Context<'a>,
+        wrapper: Handle<'a, JsObject>,
+        borrow: fn(&'a BoxContentsFor<Borrow::Target>) -> Borrow,
+    ) -> NeonResult<Self> {
+        let boxed_value: Handle<'a, DefaultJsBox<BoxContentsFor<Borrow::Target>>> =
+            Object::get(*wrapper, cx, NATIVE_HANDLE_PROPERTY)?.downcast_or_throw(cx)?;
+        let box_contents: &BoxContentsFor<Borrow::Target> = &*boxed_value;
+        // FIXME: Workaround for https://github.com/neon-bindings/neon/issues/678
+        // The lifetime of the boxed contents is necessarily longer than the lifetime of any handles
+        // referring to it, i.e. longer than 'a. However, Deref'ing a Handle can only give us a
+        // reference whose lifetime matches a *particular* handle. Therefore, we unsafely (in the
+        // compiler sense) extend the lifetime to be the lifetime of the context, as given by the
+        // Handle. (We also know the object can't move because the compiler can't know how many JS
+        // references there are referring to the JsBox, any of which another thread could
+        // be dereferencing.)
+        let contents_with_extended_lifetime: &'a BoxContentsFor<Borrow::Target> =
+            unsafe { extend_lifetime(box_contents) };
+        Ok(Self {
+            _owner: boxed_value,
```

Comment:
_js_boxed_bridge_handle: js_boxed_bridge_handle
borrowed_bridge_handle: borrowed_js_box_contents_with_extended_lifetime

Would be easier to understand and read here.